src/main/java/org/cascadebot/cascadebot/commands/moderation/PurgeAttachmentsSubCommand.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2019 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.commands.moderation;
+
+import net.dv8tion.jda.api.entities.Member;
+import org.cascadebot.cascadebot.commandmeta.*;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+import org.cascadebot.cascadebot.utils.PurgeUtils;
+
+
+public class PurgeAttachmentsSubCommand implements ISubCommand {
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        if (context.getArgs().length == 0) {
+            context.getUIMessaging().replyUsage();
+            return;
+        }
+        PurgeUtils.Purge(context, PurgeUtils.Criteria.ATTACHMENT, context.getArgAsInteger(1), context.getArg(0));
```

Comment:
For the syntax of the commands with search criteria you should have the search criteria be at the end of message and do `context.getMessage(1)`, but only if the args gets to the point where the message can exist. so
```
String search = null;
if (context.getArgs().length > 1) {
    search = context.getMessage(1);
}
```
Right now you have it where if they don't enter a value to delete they'll run into errors as the have the value second, and if they have more then one word they search for they'll run into issues.
(PS; we dropped the hole `"` system where messages with "thing stuff" would be passed in as one arg of `thing stuff`, witch is why any getting of multiple worlds generally needs to be at the end of the arguments)

Reply:
Okay then.